apiserver/facades/controller/caasunitprovisioner/provisioner.go

```diff
@@ -226,6 +227,45 @@ func (f *Facade) applicationScale(tagString string) (int, error) {
 	return app.GetScale(), nil
 }
 
+// DeploymentMode returns the deployment mode of the given applications' charms.
+func (f *Facade) DeploymentMode(args params.Entities) (params.StringResults, error) {
+	results := params.StringResults{
+		Results: make([]params.StringResult, len(args.Entities)),
+	}
+	for i, arg := range args.Entities {
+		mode, err := f.applicationDeploymentMode(arg.Tag)
+		if err != nil {
+			results.Results[i].Error = common.ServerError(err)
+			continue
+		}
+		results.Results[i].Result = mode
+	}
+	return results, nil
+}
+
+func (f *Facade) applicationDeploymentMode(tagString string) (string, error) {
+	appTag, err := names.ParseApplicationTag(tagString)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	app, err := f.state.Application(appTag.Id())
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	ch, _, err := app.Charm()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	var mode charm.DeploymentMode
+	if d := ch.Meta().Deployment; d != nil {
+		mode = d.DeploymentMode
+	}
+	if mode == "" {
+		mode = charm.ModeWorkload
```

Comment:
This probably should have been handled by the charm lib

Reply:
Just being defensive. charm lib does declare that workload is default

Reply:
IMO defensive would have been to error.